### ISSUE TITLE
loss in ModelCompileConfig doesn't accept functions

### ISSUE THREAD

**Issue by mysticatea** [tensorflow/tfjs#98]
#### TensorFlow.js version

- 0.7.0

#### Browser version

- N/A

#### Describe the problem or feature request

> String (name of objective function) or objective function.
> https://github.com/tensorflow/tfjs-layers/blob/4905b2b5106f33df2ff80c091b655703ec016094/src/engine/training.ts#L591

The `loss` property in `ModelCompileConfig` looks to say accepting objective functions, but the implementation accepts only strings.

#### Code to reproduce the bug / link to feature request

- https://github.com/tensorflow/tfjs-layers/blob/4905b2b5106f33df2ff80c091b655703ec016094/src/engine/training.ts#L597

**Comment by nsthorat** [comment]
It looks like Keras supports losses as a non-string argument: https://keras.io/losses/#usage-of-loss-functions.

We can easily extend this to take a function that allows you to define a loss at the ops-level.

Thanks for filing the issue.

**Comment by caisq** [comment]
Confirming this is fixed now.
See codepen: https://codepen.io/caisq/pen/QZYZEZ
